1046 update only the function fix

zzggo opened the pull request:
## Related Issue

Closes #1046 

<!-- If this PR addresses an issue, link it here  -->

## Summary of Changes

Check the routing of the function if it's connected to flow wallet. Doesn't include the script injection update in this one.

## Need Regression Testing

<!-- Indicate whether this PR requires regression testing and why. -->

- [x] Yes
- [ ] No

## Risk Assessment

<!-- Assess the risk level of this PR:
- Low: Minimal impact, straightforward changes.
- Medium: Potential for some edge cases or indirect effects.
- High: Could affect critical functionality or many users.
-->

- [ ] Low
- [ ] Medium
- [x] High

## Additional Notes

<!-- Add any other context or comments about the PR here -->

## Screenshots (if applicable)

<!-- Attach any screenshots that help explain your changes -->

zzggo committed:
fixed: check the routing of method and route it to flow if connected.

Closes #1040

Closes #1046

zzggo committed:
fixed: connection conflict

Closes #1040

Closes #1046

zzggo committed:
fixed: skip popup if it ethChainid call

Closes #1046

Commit:
feat: move to async

Closes #1046

tombeckenham reviewed (commented):
I'm adding Hao into this to have a look. I think it works for now without us having to completely rewrite the content script. I am still confused _why_ it works as I would have thought the methods on window.ethereum wouldn't be called if the dApp properly supports eip-6963

lmcmz reviewed (commented):
I don't think this is the right fix, especially we can't explain why you think it will work with those changes.
Please also test multiple ETH wallet extension senario.
For example install FW, MM, Rainbow, rabby all.
https://chromewebstore.google.com/detail/rainbow/opfgelmcmbiajamepnmloijbpoleiama

Commit:
fixed: provide extension id

Closes #1046

tombeckenham committed:
Update build configurations and package dependencies

- Changed module and target in tsconfig.json to ESNext for improved compatibility.
- Updated webpack configuration to use string literals for fallback modules instead of require.resolve.
- Modified prepareManifest.ts to read package.json dynamically, avoiding ESM import issues.
- Updated package.json to include ts-node to build webpack
- Adjusted pnpm-lock.yaml to reflect updated package versions and dependencies.

Closes #1050

Commit:
Merge remote-tracking branch 'origin/1050-task-correct-webpack-for-github-actions' into 1046-update-only-the-function-fix

Closes #1046
